langchain-core/src/context.ts

```diff
@@ -129,3 +133,25 @@ export function getContextVariable<T = any>(name: PropertyKey): T | undefined {
   const runTree = AsyncLocalStorageProviderSingleton.getInstance().getStore();
   return runTree?.[_CONTEXT_VARIABLES_KEY]?.[name];
 }
+
+const LC_CONFIGURE_HOOKS_KEY = Symbol("lc:configure_hooks");
+
+export const getConfigureHooks = () =>
+  getContextVariable<ConfigureHook[]>(LC_CONFIGURE_HOOKS_KEY) || [];
+
+export const registerConfigureHook = (config: ConfigureHook) => {
+  if (config.envVar && !config.handlerClass) {
+    throw new Error(
+      "If envVar is set, handlerClass must also be set to a non-None value."
+    );
+  }
+
+  setContextVariable(LC_CONFIGURE_HOOKS_KEY, [...getConfigureHooks(), config]);
+};
+
+type ConfigureHook = {
+  contextVar: string;
+  inheritable: boolean;
+  handlerClass?: BaseCallbackHandler | CallbackHandlerMethods;
```

Comment:
I don't think we want to accept a raw methods object here because then we can't dedupe by `name` later?